app/controllers/miq_ae_class_controller.rb

```diff
@@ -2819,13 +2824,19 @@ def get_method_node_info(node_id)
     set_right_cell_text(x_node, @record)
   end
 
-  def fetch_manager_name(ansible_template_id)
+  def fetch_manager_name(ansible_template_id, klass)
     return nil if ansible_template_id.blank?
 
-    ManageIQ::Providers::ExternalAutomationManager::ConfigurationScript.find_by(:id => ansible_template_id)&.manager&.name
+    klass.find_by(:id => ansible_template_id)&.manager&.name
   end
 
   def fetch_playbook_details(record)
+    template_class = if record.location == "ansible_workflow_template"
+                       ManageIQ::Providers::ExternalAutomationManager::ConfigurationWorkflow
+                     else
+                       ManageIQ::Providers::ExternalAutomationManager::ConfigurationScript
+                     end
```

Comment:
This duplicates the logic from line 1038, so maybe it's better to move into a shared method.

Reply:
Maybe a `fetch_manager` method or something to kind of align with the `fetch_manager_name` method naming-wise.

Reply:
Or perhaps a `template_class` method.